src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -263,18 +272,60 @@ public void renameFileToName(String targetFileName) {
         }
     }
 
-    private void performRenameWithConflictCheck(String targetFileName) {
+    public void performRenameWithConflictCheck(String targetFileName) {
         Optional<Path> existingFile = linkedFileHandler.findExistingFile(linkedFile, entry, targetFileName);
         boolean overwriteFile = false;
 
         if (existingFile.isPresent()) {
-            overwriteFile = dialogService.showConfirmationDialogAndWait(
-                    Localization.lang("File exists"),
-                    Localization.lang("'%0' exists. Overwrite file?", targetFileName),
-                    Localization.lang("Overwrite"));
+            Path existingFileVal = existingFile.get();
+            // show when hovering over "Keep both" button
+            String suggestedFileName = FileNameUniqueness.getNonOverWritingFileName(
+                existingFileVal.getParent(),
+                existingFileVal.getFileName().toString()
+            );
+
+            Label contentLabel = new Label(Localization.lang("Target file name: \n'%0'", targetFileName));
+            contentLabel.setWrapText(true);
+            DialogPane dialogPane = new DialogPane();
+            dialogPane.setPrefSize(700, 200);
+            dialogPane.setContent(contentLabel);
+
+            ButtonType overrideButton = new ButtonType("Override", ButtonBar.ButtonData.OTHER);
+            ButtonType keepBothButton = new ButtonType("Keep both", ButtonBar.ButtonData.OTHER);
+            ButtonType provideAltFileNameButton = new ButtonType("Provide alternative file name", ButtonBar.ButtonData.OTHER);
+
+            dialogPane.getButtonTypes().addAll(overrideButton, keepBothButton, provideAltFileNameButton);
+
+            dialogPane.sceneProperty().addListener((obs, oldScene, newScene) -> {
+                if (newScene != null) {
+                    Platform.runLater(() -> {
+                        Button btnKeepBoth = (Button) dialogPane.lookupButton(keepBothButton);
+                        if (btnKeepBoth != null) {
+                            Tooltip tooltip = new Tooltip(Localization.lang("New name: %0", suggestedFileName));
+                            tooltip.setShowDelay(Duration.millis(100));
+                            tooltip.setStyle("-fx-max-width: 20px; -fx-wrap-text: true; -fx-background-color: #FFFFE0;");
```

Comment:
px values dont scale. use em
dont manually set colors. use constants from base.css

Reply:
put the whole style into base.css and only set a style class for this tooltip